### PR TITLE
Data convertor timestamp support and UT refactor by type

### PR DESCRIPTION
**Pull request by yyang52**
Add support to timestamp round trip convert: 
Cider to velox convert: pass dimen from cider side to indicate the precision scale, convert to velox timestamp which is represented by seconds and nanoseconds (may have precision loss here). 
Velox to cider: as cider has different dimension for timestamp, for now convert to nanoseconds to avoid precision loss.

### CODE REVIEW

**Comment by yyang52** [comment]
@Intel-bigdata/bdf-cider-engine @jikunshang

File: velox/omnisci/RawDataConvertor.cpp [hunk 1]
```diff
@@ -247,28 +269,43 @@ VectorPtr toVeloxImpl<TypeKind::TIMESTAMP>(
...
       result->setNull(pos, true);
     } else {
-      auto microseconds = srcValues[pos];
-      // TODO: need to get precision info from omnisci
-      flatResult->set(
-          pos,
-          Timestamp(microseconds / 1000000, (microseconds % 1000000) * 1000));
+      auto timeValue = srcValues[pos];
+      switch (dimen) {
+        case 0:
```
Comment (jikunshang) [inline velox/omnisci/RawDataConvertor.cpp:283]: what's 0,3,6,9 meaning here?

File: velox/omnisci/RawDataConvertor.cpp [hunk 2]
```diff
@@ -247,28 +269,43 @@ VectorPtr toVeloxImpl<TypeKind::TIMESTAMP>(
...
     if (srcValues[pos] == std::numeric_limits<int64_t>::min()) {
       result->setNull(pos, true);
     } else {
-      auto microseconds = srcValues[pos];
-      // TODO: need to get precision info from omnisci
-      flatResult->set(
-          pos,
-          Timestamp(microseconds / 1000000, (microseconds % 1000000) * 1000));
+      auto timeValue = srcValues[pos];
+      switch (dimen) {
```
Comment (jikunshang) [inline velox/omnisci/RawDataConvertor.cpp:282]: `dimen` will be fixed in one function call, move it out of for loop to be more efficient?

File: velox/omnisci/tests/ResultConvertTest.cpp [hunk 3]
```diff
@@ -36,8 +36,8 @@ TEST_F(ResultConvertTest, VeloxToCiderDirectConvert) {
       DataConvertor::create(CONVERT_TYPE::DIRECT);
   int numRows = 10;
   auto rowType =
-      ROW({"col_0", "col_1", "col_2", "col_3"},
-          {INTEGER(), BIGINT(), DOUBLE(), BOOLEAN()});
+      ROW({"col_0", "col_1", "col_2", "col_3", "col_4"},
```
Comment (jikunshang) [inline velox/omnisci/tests/ResultConvertTest.cpp:39]: I suggest to separate each type in different test cases.

**Review by jikunshang — COMMENTED** [review]
can you describe how you did this conversion briefly?

File: velox/omnisci/RawDataConvertor.cpp [hunk 4]
```diff
@@ -247,28 +269,43 @@ VectorPtr toVeloxImpl<TypeKind::TIMESTAMP>(
...
+        case 0:
+          flatResult->set(pos, Timestamp(timeValue, 0));
+          break;
+        case 3:
+          flatResult->set(
+              pos,
+              Timestamp(
+                  timeValue / kMilliSecsPerSec,
+                  (timeValue % kMilliSecsPerSec) * 1000000));
```
Comment (yma11) [inline velox/omnisci/RawDataConvertor.cpp:291]: use kMicroSecsPerSec instead of 1000000 here?

File: velox/omnisci/RawDataConvertor.cpp (hunk 1 above)
Comment (yyang52) [reply velox/omnisci/RawDataConvertor.cpp:283]: omnisci dimension for second, millisecond, microsecond, nanosecond

**Review by jikunshang — COMMENTED** [review]
Please change PR title if necessary and illustrate what dime is in description or code.

File: velox/omnisci/RawDataConvertor.cpp [hunk 5]
```diff
@@ -276,6 +276,21 @@ VectorPtr toVeloxImpl<TypeKind::VARBINARY>(
   VELOX_NYI(" {} conversion is not supported yet");
 }
 
+std::tuple<int64_t, int64_t> calculateScale(int32_t dimen) {
+  switch (dimen) {
+    case CIDER_DIMEN::SECOND:
+      return {1, 1};
```
Comment (jikunshang) [inline velox/omnisci/RawDataConvertor.cpp:282]: `1 ` here could be some constant rather than magic number?

**Comment by jikunshang** [comment]
LGTM.

**Review by jikunshang — APPROVED** [review]

File: velox/omnisci/CiderNullValues.h [hunk 6]
```diff
@@ -25,6 +25,7 @@
 #define CONSTEXPR constexpr
 
 using namespace facebook::velox;
+
```
Comment (winningsix) [inline velox/omnisci/CiderNullValues.h:28]: revert this.

**Review by winningsix — APPROVED** [review]

File: velox/omnisci/RawDataConvertor.cpp [hunk 7]
```diff
@@ -15,7 +15,7 @@
 #include "velox/vector/FlatVector.h"
 
 namespace facebook::velox::cider {
-
+
```
[the remark this answers is not in the document]
Comment (yyang52) [reply velox/omnisci/RawDataConvertor.cpp]: revert which? the related commit?

[the remark this answers is not in the document]
Comment (winningsix) [reply velox/omnisci/RawDataConvertor.cpp]: unrelated code changes. A new space added here. You can do it in other PR.

[the remark this answers is not in the document]
Comment (yyang52) [reply velox/omnisci/RawDataConvertor.cpp]: ok, will create a new one to address that

[the remark this answers is not in the document]
Comment (winningsix) [reply velox/omnisci/RawDataConvertor.cpp]: You can do it in your next PR. No need to create a single PR just for such minor comments :)

**Pull request merged**